src/library.tests/RemoteContainerConnectionDetailsResolverTests.cs

```diff
@@ -38,7 +37,7 @@ public RemoteContainerConnectionDetailsResolverTests()
         public async void ResolveConnectionDetails_RestoreCheck(int numServices, bool hasRestore, string phase)
         {
             // Set up
-            ConfigurePods(numPods: numServices, namingFunction: (i) => hasRestore && i == 0 ? $"myapp-restore-{i}" : $"myapp-{i}", phaseFunction: (i) => i == 0 ? phase : "Running");
+            ConfigurePods(numPods: numServices, namingFunction: (i) => hasRestore && i == 0 ? $"myapp-restore-{i}" : $"myapp-{i}", phaseFunction: (i) => i == 0 ? phase : "Running", "Job", "linux");

```

Comment:
💡 Idea/thoughts please: we could have an enum of state and have these `running et. al.` as follows, will be consistent and possible just fyi to casing inconsistency as well 

```*.cs
enum phase 
{
  Running,
  Job,
  Linux
}
```

Reply:
Sure will add them as Enum for test class file.

Reply:
This threw me at first. "Job" and "linux" are actually different arguments to the `ConfigurePods` function, but at first glance they look like some kind of multi-return type from `phaseFunction`.

I think named arguments would help a lot here, even though they'd make the function call really long. To deal with that, you could just wrap the lines (one arg per line), or maybe create a struct/record type for the parameters (which you could maybe argue is overkill).

Reply:
I'm not totally sure if Enums will work here, because we might not have a constrained set of possible values for `kind` and `nodeSelector`.

Reply:
good point @peterbom. I was thinking to create Enum's for Phase - Running, Pending etc.. similarly for nodeSelector - linux, windows etc.. I think Tats would have meant the same.